### DISCUSSION TITLE
Did Not Connect: Potential Security Issue (Error code: SEC_ERROR_UNKNOWN_ISSUER) with Let's encrypt cert

### DISCUSSION THREAD

**Discussion by Suncatcher** [Ylianst/MeshCentral#5988] [Q&A] [answered]
Hi everybody,
my config with built-in ertificate that worked like a charm for a while suddenly stopped working. I see this error and it doesn't allow me to pass thru
![image](https://github.com/Ylianst/MeshCentral/assets/6388034/1a2e8544-764f-44ed-841d-4ff3555a3859)

From my perspective nothing has changed since the [last issue](https://github.com/Ylianst/MeshCentral/discussions/5900) was created, so the configuration is following:

```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "XXXXXXXXXXXXX",
    "WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "_port": 443,
    "_aliasPort": 443,
    "_redirPort": 80,
    "_redirAliasPort": 80,
    "mpsport": 0
  },
  "domains": {
    "": {
      "_title": "MyServer",
      "_title2": "Servername",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true
    }
  },
  "letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "XXXXXXXXXXX",
    "names": "XXXXXXXXXXX",
    "skipChallengeVerification": true,
    "production": true
  }
}
```
I checked just now and even the IPv6 of my EC2 instance didn't change, it is effectively the same as month before /below is dummy/: 
[https://[23c0:90c3:7b9b:3033:24b0:781a:b252:ccf6]/](https://[23c0:90c3:7b9b:3033:24b0:781a:b252:ccf6]/)
and I can successfully open the Mesh by IP address like this, though only after the following warning
![image](https://github.com/Ylianst/MeshCentral/assets/6388034/4a7de215-45ef-411e-8a55-4196cc913b30)

Any suggestion why it was broken?

**Comment by si458** [comment]
If u click `view certificate` what does the dn/cn say? 
It should be the dns name u set in cert?
Also check who issued the certificate
Share pic if u can

**Reply by Suncatcher** [reply]
![image](https://github.com/Ylianst/MeshCentral/assets/6388034/ef6f6804-ce3f-403d-962c-cd7c086fa82c)
![image](https://github.com/Ylianst/MeshCentral/assets/6388034/10bd3326-82e7-4808-81d3-54cf7ceee1e3)
Does it say you anything?

**Reply by Suncatcher** [reply]
> It should be the dns name u set in cert?

DNS name that I distorted on the screen above, corresponds to my domain and listed in two variants in the certificate: 

pure `domain.com` and `http://domain.com`, no HTTPS version. Should it be https listed?

**Reply by Suncatcher** [reply]
@si458 could you please help?

**Reply by si458** [reply]
so it seems like its using the self-signed certs instead of your letsencypt cert.
how do you run/start meshcentral?
when you start it with `node node_modules/meshcentral` is should show logs, what do they say?
is the any messages on the screen in `My Server` tab?
what about clicking the `show server error logs` in the `my server` tab ?

**Reply by Suncatcher** [reply]
> what about clicking the show server error logs in the my server tab ?

no errors in `show server error logs`

![image](https://github.com/Ylianst/MeshCentral/assets/6388034/012dc792-9911-4518-af03-e4dc64495382)

but i see this
![image](https://github.com/Ylianst/MeshCentral/assets/6388034/b6bad60c-f5f3-4a31-92d2-89062690ffea)

what happened and how to fix it? what config to check?

**Reply by Suncatcher** [reply]
> how do you run/start meshcentral?

like `systemctl enable MeshCentral.service`

**Reply by si458** [reply]
right so the `names` section in `letsencrypt` must be wrong,
OR
the meshcentral server cannot do a DNS lookup for the DNS name.

you should only have a single dns name in there, unless you want multiple,
so `mc.mydomain.com` or `mc1.mydomain.com,mc2.mydomain.com`

```
  "letsEncrypt": {
    "email": "simon@mydomain.com",
    "names": "mc.mydomain.com",
    "skipChallengeVerification": true,
    "production": true
  },
```

**Reply by Suncatcher** [reply]
> you should only have a single dns name in there

I have only single domain in that field in the format `domain.com`

I checked in https://www.whatismyip.com/dns-lookup/ and it seems the problem is with my domain. It fetches fine the NS records when I specify domain with HTTPS
![image](https://github.com/Ylianst/MeshCentral/assets/6388034/e69eb789-0432-4538-bf52-c69f127cde45)

but doesn't see them when putting bare domain `domain.com`
![image](https://github.com/Ylianst/MeshCentral/assets/6388034/39bec47e-0ae7-41be-9b85-c948769b4f51)

any idea what I am missing? Should I put HTTPS format into let's encrypt field? My domain is registered thru Cloudflare registrar

**Reply by si458** [reply]
the only other thing you can try is setting `nochecks: true` inside your `letsencrypt` and restart your server,
this will bypass ALL safety checks like email MX records and A or AAAA lookups! 
but use this at your own risk as letsencrypt could block you for multiple failed attempts to get a certificate due to verification!
so best set `production: false` as well as `nochecks: true` and see if you get the test letsencrypt cert,
if you do, then set production back to true

**Reply by si458** [reply]
you need to check your server can do the DNS lookup of your domain, not an external provider.
`nslookup mydomain.com` or `dig mydomain.com AAAA`

**Reply by Suncatcher** [reply]
> nslookup mydomain.com

```
[ec2-user@ip-172-XX-XXX-XX meshcentral-data]$ nslookup domain.com
Server:         172-XX-XXX-XX
Address:        172-XX-XXX-XX#53

Non-authoritative answer:
Name:   domain.com
Address: XX00:XXXX:24:XXXX:XXXX:XXXX:XXXX:2XX1 <--- **my IPv6 address**
```

> dig mydomain.com AAAA

```
[ec2-user@ip-172-XX-XXX-XX meshcentral-data]$ dig domain.com AAAA

; <<>> DiG 9.16.42-RH <<>> domain.com AAAA
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 29347
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;domain.com.                  IN      AAAA

;; ANSWER SECTION:
domain.com.           251     IN      AAAA    XX00:XXXX:24:XXXX:XXXX:XXXX:XXXX:2XX1

;; Query time: 0 msec
;; SERVER: 172-XX-XXX-XX#53(172-XX-XXX-XX)
;; WHEN: Sat Apr 06 22:17:03 UTC 2024
```

Anything looks wrong?

**Reply by si458** [reply]
i have a feeling its because you are using IPv6 ONLY in the dns as you have no ipv4 address.
i will look into it as i now have ipv6 in our rack so i can do more testing!
try my suggestion above adding the `nochecks: true` to `letsencrypt` https://github.com/Ylianst/MeshCentral/discussions/5988#discussioncomment-9032027
remember to restart meshcentral after you set it `systemctl restart meshcentral.service`

**Reply by Suncatcher** [reply]
> i have a feeling its because you are using IPv6 ONLY in the dns as you have no ipv4 address

correct, that is my setup. Is that a problem that was never tested by MeshCentral or is it just my wrong setup?

**Reply by si458** [reply]
yes it will be a problem that was never tested,
you have to remember @Ylianst did all this on his own along with @krayon007 
so its hard to test every single environment!
but give the `nochecks: true` a try as that will completely bypass the checks for letsencrypt

**Reply by Suncatcher** [reply, marked as answer]
> try my suggestion above adding the nochecks: true to letsencrypt

that solved the problem, thanks. But I'm concerned about the longstanding consequences you mentioned: I don't want my domain to be banned by Let's Encrypt for multiple wrong attempts

**Reply by si458** [reply]
dont worry, if you can open a bug report with all the details (follow template 🙏),
ill put it on my to-do list and have a look during the week!

**Reply by si458** [reply]
yeh its because your dns name only have an AAAA record, so the dns fails, as it currently only looks for an A record!
```
> dns.resolve('meshcentral.mydomain.com', function(err,records){ console.log(err,records); });
QueryReqWrap {
  bindingName: 'queryA',
  callback: [Function (anonymous)],
  hostname: 'meshcentral.mydomain.com',
  oncomplete: [Function: onresolve],
  ttl: false
}
> null [ '1.2.3.4' ]
```

**Reply by si458** [reply]
actually just fixed it! very simple! 1 line! dont worry about a bug report! https://github.com/Ylianst/MeshCentral/commit/31ebb21e0be8c31b424db42d7902fdddd64f549e

**Reply by Suncatcher** [reply]
you are just awesome, bro! so this means that I can remove that `nocheck` option after I upgrade the mesh to the latest version from Git source? BTW, what is the update policy of MeshCentral? This fix will be included in the next major version?

**Reply by si458** [reply]
yes, when we release a new meshcentral update it will be included and you can remove `nocheck`
1.1.23 will be the next release, which SHOULD be due next month maybe? not sure yet. @Ylianst does the releases when he gets chance and every so often.
i use to aim for a release every month, but i no it call be every 2 months say.
but if ur impatient (like me), then just `git clone` the master branch for all the latest features/bug fixes!
